xds/internal/clients/xdsclient/authority.go

```diff
@@ -955,3 +955,14 @@ func serviceStatusToProto(serviceStatus xdsresource.ServiceStatus) v3adminpb.Cli
 		return v3adminpb.ClientResourceStatus_UNKNOWN
 	}
 }
+
+func (a *authority) resourceWatchStateForTesting(rType ResourceType, resourceName string) (state xdsresource.ResourceWatchState, err error) {
+	done := make(chan struct{})
+	a.xdsClientSerializer.ScheduleOr(func(context.Context) {
+		state, err = a.activeXDSChannel.channel.ads.adsResourceWatchStateForTesting(rType, resourceName)
+		close(done)
+	}, func() { close(done) })
+	<-done
```

Comment:
Should the scheduler have a function like:

```go
// Schedules f and waits for it to be executed.
func (cs *CallbackSerializer) ScheduleAndWait(f func(ctx context.Context))
```

Maybe even something nice with generics:

```go
// Schedules f and waits for it to be executed.  Returns what T returned or nil and
// ErrSerializerClosedBeforeItCouldRunFOrWhatever.
func (cs *CallbackSerializer) ScheduleAndWait[T any](f func(ctx context.Context) T) (T, error)
```

(I think we have this pattern multiple times?)

Just an idea to discuss, not for this PR.

Reply:
Yes, we have this pattern in a whole bunch of places, and would definitely make sense to add a method like what you've suggested here to the serializer.

Reply:
Filed https://github.com/grpc/grpc-go/issues/8501 to track this.